steps/jira-issue-update-build/0.1.3/step.yml

```diff
@@ -0,0 +1,82 @@
+  it looks for task keys using a predefined format (for example, [ABCD-1234]). Then
+  using JIRA API, it updates custom fields of these tasks with current build number.
+  \n\nAdditionally step can keep ticket history from failed/aborted builds. It uses
+  Bitrise API to gather information about the commits \nfrom all aborted or failed
+  builds preceding the current one.\n\n### Useful links\n- [About this step](https://www.holdapp.com/blog/bitrise-tests-made-easier-with-jira-build-step)"
+website: https://github.com/Holdapp/bitrise-step-jira-build
+source_code_url: https://github.com/Holdapp/bitrise-step-jira-build
+support_url: https://github.com/Holdapp/bitrise-step-jira-build/issues
+published_at: 2021-02-02T17:08:57.274627+01:00
+source:
+  git: https://github.com/Holdapp/bitrise-step-jira-build.git
+  commit: 21663a6bb7ba0b8609778120ea1f12f89171f58b
+host_os_tags:
+- osx-10.10
+- ubuntu-16.04
+type_tags:
+- notification
+toolkit:
+  go:
+    package_name: github.com/Holdapp/bitrise-step-jira-build
+deps:
+  brew:
+  - name: git
+  - name: libgit2
+  apt_get:
+  - name: git
+  - name: libgit2-dev
+is_requires_admin_user: true
+is_always_run: false
+is_skippable: true
+inputs:
+- APP_VERSION: null
+  opts:
+    description: "If you develop an iOS project, you can extract the version number
+      from Info.plist of the main target. It’s called `CFBundleShortVersionString`.
+      \nAlternatively, you may use a custom step that can do it for you: [bitrise-step-xcode-build-version](https://github.com/nodes-ios/bitrise-step-xcode-build-version)"
+    is_required: true
+    summary: App version (for example 1.0.0)
+    title: App version
+- JIRA_HOST: null
+  opts:
+    is_required: true
+    summary: Your JIRA instance URL (e.g. https://company.atlassian.net)
+    title: JIRA host
+- JIRA_CUSTOM_FIELD_ID: null
+  opts:
+    description: "You can read how to find it [here](https://confluence.atlassian.com/jirakb/how-to-find-id-for-custom-field-s-744522503.html).
+      \nAnd if you don’t have a custom field yet, check out this [guide](https://confluence.atlassian.com/adminjiraserver/adding-a-custom-field-938847222.html)
+      that explains how to create it."
+    is_required: true
+    summary: Custom field id for the build number (integer)
+    title: JIRA Custom Field ID
+- JIRA_USERNAME: null
+  opts:
+    is_required: true
+    summary: User (or bot) username
+    title: JIRA username
+- JIRA_ACCESS_TOKEN: null
+  opts:
+    is_required: true
+    is_sensitive: true
+    summary: User (or bot) password or API token (you can generate it [here](https://confluence.atlassian.com/cloud/api-tokens-938839638.html)).
+    title: JIRA access token
+- JIRA_ISSUE_PATTERN: ([A-Z]{1,10}-[0-9]+)
+  opts:
+    is_required: true
+    summary: A regular expression for matching issue keys in commit messages. For
+      example, ([A-Z]{1,10}-[0-9]+)
+    title: Regex pattern used to identify issue keys from commit messages
+- BITRISE_API_TOKEN: null
```

Comment:
What does this step use the Bitrise API for? Don't you have all build related info in the build itself? I mean these default env vars (including the build number): https://devcenter.bitrise.io/builds/available-environment-variables/

Reply:
Step is using Bitrise API, to keep JIRA issues from the commits of aborted/failed builds preceding the current one. This way if for some reason build failed, info about issues it resolved would not be lost.